privval/grpc/client.go

```diff
@@ -0,0 +1,104 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+
+	grpc "google.golang.org/grpc"
+	"google.golang.org/grpc/status"
+
+	"github.com/tendermint/tendermint/crypto"
+	cryptoenc "github.com/tendermint/tendermint/crypto/encoding"
+	"github.com/tendermint/tendermint/libs/log"
+	privvalproto "github.com/tendermint/tendermint/proto/tendermint/privval"
+	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
+	"github.com/tendermint/tendermint/types"
+)
+
+// SignerClient implements PrivValidator.
+// Handles remote validator connections that provide signing services
+type SignerClient struct {
+	ctx    context.Context
```

Comment:
Contexts should generally never be stored in a struct, but instead be passed for each call -- see [context docs](https://golang.org/pkg/context/).

Reply:
Yes, this is a workaround to avoid breaking the privval interface. If we are fine with breaking the interface I can make the change. I agree it would be better.

Reply:
If we don't want to break the interface then we may as well just remove this and generate a new context for each call, since there is no way for a client to provide a call-scoped context anyway (which is the main purpose of contexts).

Reply:
Migrated to using contexts with a deadline. Do you think this should be configurable?